jsf-ri/src/main/java/com/sun/faces/spi/AnnotationProviderFactory.java

```diff
@@ -70,12 +70,27 @@ public static AnnotationProvider createAnnotationProvider(ServletContext sc) {
         String[] services = ServiceFactoryUtils.getServiceEntries(ANNOTATION_PROVIDER_SERVICE_KEY);
         if (services.length > 0) {
             // only use the first entry...
-            Object provider = ServiceFactoryUtils.getProviderFromEntry(services[0],
-                new Class[] { ServletContext.class, AnnotationProvider.class }, new Object[] { sc , annotationProvider });
+            Object provider = null;
+            try {
+                // try two arguments constructor
+                provider = ServiceFactoryUtils.getProviderFromEntry(services[0],
+                    new Class[] { ServletContext.class, AnnotationProvider.class }, new Object[] { sc , annotationProvider });
+            } catch (FacesException e) {
+                // rethrow serious problem
+                throw e;
```

Comment:
Since ```AnnotationProviderFactory#createAnnotationProvider``` is public API, we should be careful not to change the existing behaviour of this method. Until now, this method would simply ignore any exceptions that occur when attempting to invoke the two-argument constructor (this includes the ```NoSuchMethodException``` as well as any other exceptions that occur even if the constructor is found). So, the ```FacesException``` here should just be ignored.